emily_cron/app/services/deposit_processor.py

```diff
@@ -0,0 +1,155 @@
+            logger.info(f"Marking transaction {tx.bitcoin_txid} with expired locktime as FAILED")
+            updates.append(
+                DepositUpdate(
+                    bitcoin_txid=tx.bitcoin_txid,
+                    bitcoin_tx_output_index=tx.bitcoin_tx_output_index,
+                    last_update_height=stacks_chaintip.height,
+                    last_update_block_hash=stacks_chaintip.hash,
+                    status=RequestStatus.FAILED.value,
+                    status_message=f"Locktime expired at height {bitcoin_chaintip.height}",
+                )
+            )
+
+        return updates
+
+
+    def update_deposits(self) -> None:
+        """Update deposit statuses.
+
+        This is the main entry point for the deposit processor.
+        It fetches deposits, enriches them, processes them, and updates their status.
+        """
+        logger.info("Running deposit status update job")
+
+        # Get current blockchain state
+        bitcoin_chaintip = MempoolAPI.get_bitcoin_block_at()
+        stacks_chaintip = HiroAPI.get_stacks_block()
+
+        logger.info(f"Bitcoin chain tip: {bitcoin_chaintip}")
+        logger.info(f"Stacks chain tip: {stacks_chaintip}")
+
+        # Fetch pending and accepted deposits
+        pending_deposits = PublicEmilyAPI.fetch_deposits(RequestStatus.PENDING)
+        accepted_deposits = PublicEmilyAPI.fetch_deposits(RequestStatus.ACCEPTED)
+
+        # Enrich deposits with additional transaction data
+        enriched_deposits = self._enrich_deposits(chain(pending_deposits, accepted_deposits))
+
+        # Process deposits and collect updates
+        updates = []
+
+        # Process transactions with expired locktime
+        locktime_updates = self.process_expired_locktime(
+            enriched_deposits,
+            bitcoin_chaintip,
+            stacks_chaintip,
+        )
+        updates.extend(locktime_updates)
+
+        # Apply updates
+        if updates:
+            logger.info(f"Updating {len(updates)} deposit statuses")
+            PrivateEmilyAPI.update_deposits(updates)
+        else:
+            logger.info("No deposit updates needed")
+
+        logger.info("Deposit status update job completed")
+
+
+    def _enrich_deposits(self, deposits: Iterable[DepositInfo]) -> list[EnrichedDepositInfo]:
+        """Fetch transaction details and enrich deposit info.
+
+        Args:
+            deposits: Iterable of DepositInfo objects
+
+        Returns:
+            list[EnrichedDepositInfo]: List of enriched deposit information
+        """
+        transaction_details = []
+        for deposit in deposits:
+            tx_data = MempoolAPI.get_bitcoin_transaction(deposit.bitcoin_txid)
+
+            if not tx_data:
+                transaction_details.append(EnrichedDepositInfo.from_missing(deposit))
+                continue
+
+            spending_outputs = {
+                vout.get("scriptpubkey_address", "Unknown"): vout.get("value", 0)
```

Comment:
Both here and below: I'm not sure if default some of those is safe. What about logging and ignoring the ones missing the required ones?